### PR TITLE
Initial softmax, softmin, batch norm 2d implementation

### PR DESCRIPTION
**Pull request by alifahrri**

### CODE REVIEW

**Comment by alifahrri** [comment]
[implement batchnorm 2d](https://trello.com/c/qeYOjqMF/26-implement-batchnorm-2d)

**Comment by alifahrri** [comment]
[[nn] implement softmax](https://trello.com/c/yY8upmWL/4-nn-implement-softmax)

**Comment by alifahrri** [comment]
[implement softmin](https://trello.com/c/RLbCcZhd/52-implement-softmin)
